### PR TITLE
Improve code security

### PR DESCRIPTION
**Pull request by GermanAizek**
@caesay, @rstarkov review code pull request and feedback me.

### CODE REVIEW

File: src/Clowd.WinNative/outline-text/OutlineText.cpp [hunk 1]
```diff
@@ -715,9 +715,9 @@ void OutlineText::RenderFontShadow(
 		}
 	}
 
-	pBmpMask->UnlockBits(&bitmapDataMask);
-	pBmpDisplay->UnlockBits(&bitmapDataDest);
-	pBmpFontBodyBackup->UnlockBits(&bitmapDataSrc);
+    if (pBmpMask) pBmpMask->UnlockBits(&bitmapDataMask);
```
Comment (caesay) [inline src/Clowd.WinNative/outline-text/OutlineText.cpp:718]: if any of these 3 locals are null, this method will crash earlier - eg. memory violation on line 652, so we probably don't need to check this here

**Review by caesay — CHANGES_REQUESTED** [review]
Hi! Thanks for submitting this change. Do you mind me asking if you use Clowd and how you found out about it? Please join our discord if you are interested to discuss any further potential changes!

There are some really good improvements in this PR, however there are also a few places that I am unsure and need clarification from you.

File: src/Clowd.WinNative/DxScreenCapture.cpp [hunk 2]
```diff
@@ -2005,7 +2005,7 @@ json rect2json(RECT& rc)
     return j;
 }
 
-System::String DxScreenCapture::SaveSession(System::String sessionDirectory, System::String createdUtc)
+System::String DxScreenCapture::SaveSession(System::String sessionDirectory, const System::String& createdUtc)
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/Clowd.WinNative/DxScreenCapture.cpp:2008]: @caesay, Yep, for sure I did not notice, I will fix it now.

**Comment by caesay** [comment]
Closing this as it requires much feedback and there has been no activity. Feel free to open a new PR when the comments have been addressed.

**Pull request closed without merging**